spec/Functions.tex

```diff
@@ -221,8 +221,7 @@ \section{Procedure Definitions}
 The \sntx{return-type} is optional and is discussed in~\rsec{Return_Types}.
 A type function may not specify a return type.
 
-The optional \sntx{where-clause} is only applicable if the function is
-generic.  It is discussed in~\rsec{Where_Expressions}.
+The optional \sntx{where-clause} is discussed in~\rsec{Where_Expressions}.
```

Comment:
I hate to say this since (a) it involves the spec and (b) you've taken this effort on from me, but I think that we need to move the current Where Expressions section from the generics chapter up to the Procedures chapter; or perhaps move most of it up to Procedures and have a section in Generics refining the concrete explanation.  You can push this back on me if you want.